src/Persistence/PersistentObjectFactory.php

```diff
@@ -290,30 +291,42 @@ protected function normalizeParameter(string $field, mixed $value): mixed
+            // handle inverse OneToOne
+            if ($relationshipMetadata && !$relationshipMetadata->isCollection) {
+                $inverseField = $relationshipMetadata->inverseField;
 
-                $inversedObject = $value->withPersistMode(
+                $value = $value->withPersistMode(
                     $this->isPersisting() ? PersistMode::NO_PERSIST_BUT_SCHEDULE_FOR_INSERT : PersistMode::WITHOUT_PERSISTING
-                )
+                );
+
+                if (($fieldType = (new \ReflectionClass(static::class()))->getProperty($field)->getType())?->allowsNull()) {
+                    $this->tempAfterInstantiate[] = static function(object $object) use ($value, $inverseField, $field) {
+                        $inverseObject = $value->create([$inverseField => $object]);
 
-                    // we need to handle the circular dependency involved by inversed one-to-one relationship:
-                    // a placeholder object is used, which will be replaced by the real object, after its instantiation
-                    ->create([
-                        $inverseField => $placeholder = (new \ReflectionClass(static::class()))->newInstanceWithoutConstructor(),
-                    ]);
+                        set($object, $field, unproxy($inverseObject, withAutoRefresh: false));
+                    };
 
-                $inversedObject = unproxy($inversedObject, withAutoRefresh: false);
+                    // we're using "force" here to avoid a potential type check in a setter
+                    return force(null);
+                } elseif (($inverseFieldType = (new \ReflectionClass($value::class()))->getProperty($inverseField)->getType())?->allowsNull()) {
+                    $inverseObject = unproxy(
+                        // we're using "force" here to avoid a potential type check in a setter
+                        $value->create([$inverseField => force(null)]),
+                        withAutoRefresh: false
+                    );
 
-                $this->tempAfterInstantiate[] = static function(object $object) use ($inversedObject, $inverseField, $pm, $placeholder) {
-                    $pm->forget($placeholder);
-                    set($inversedObject, $inverseField, $object);
-                };
+                    $this->tempAfterInstantiate[] = static function(object $object) use ($inverseObject, $inverseField) {
+                        set($inverseObject, $inverseField, $object);
+                    };
 
-                return $inversedObject;
+                    return $inverseObject;
+                } elseif ($fieldType === null || $inverseFieldType === null) {
+                    throw new \InvalidArgumentException(sprintf("Cannot handle inverse OneToOne relationship: cannot determine types of \"%s::\${$field}\" and \"%s::\${$inverseField}\", please and type to the properties.", static::class(), $value::class()));
```

Comment:
I think this is enough for now: we could actually use symfony/type-info or symfony/property-info components to read phpdoc / doctrine mapping, but I can't see a real reason to not strongly type the properties of the relationship